client/ayon_core/plugins/load/delete_old_versions.py

```diff
@@ -1,501 +1,409 @@
-#         )
-#
-#         dir_paths = {}
-#         file_paths_by_dir = collections.defaultdict(list)
-#         for repre in repres:
-#             file_path, seq_path = self.path_from_representation(
-#                 repre, anatomy
-#             )
-#             if file_path is None:
-#                 self.log.debug((
-#                     "Could not format path for represenation \"{}\""
-#                 ).format(str(repre)))
-#                 continue
-#
-#             dir_path = os.path.dirname(file_path)
-#             dir_id = None
-#             for _dir_id, _dir_path in dir_paths.items():
-#                 if _dir_path == dir_path:
-#                     dir_id = _dir_id
-#                     break
-#
-#             if dir_id is None:
-#                 dir_id = uuid.uuid4()
-#                 dir_paths[dir_id] = dir_path
-#
-#             file_paths_by_dir[dir_id].append([file_path, seq_path])
-#
-#         dir_ids_to_pop = []
-#         for dir_id, dir_path in dir_paths.items():
-#             if os.path.exists(dir_path):
-#                 continue
-#
-#             dir_ids_to_pop.append(dir_id)
-#
-#         # Pop dirs from both dictionaries
-#         for dir_id in dir_ids_to_pop:
-#             dir_paths.pop(dir_id)
-#             paths = file_paths_by_dir.pop(dir_id)
-#             # TODO report of missing directories?
-#             paths_msg = ", ".join([
-#                 "'{}'".format(path[0].replace("\\", "/")) for path in paths
-#             ])
-#             self.log.debug((
-#                 "Folder does not exist. Deleting it's files skipped: {}"
-#             ).format(paths_msg))
-#
-#         return {
-#             "dir_paths": dir_paths,
-#             "file_paths_by_dir": file_paths_by_dir,
-#             "versions": versions,
-#             "folder": folder_entity,
-#             "product": product_entity,
-#             "archive_product": versions_count == 0
-#         }
-#
-#     def main(self, project_name, data, remove_publish_folder):
-#         # Size of files.
-#         size = 0
-#         if not data:
-#             return size
-#
-#         if remove_publish_folder:
-#             size = self.delete_whole_dir_paths(data["dir_paths"].values())
-#         else:
-#             size = self.delete_only_repre_files(
-#                 data["dir_paths"], data["file_paths_by_dir"]
-#             )
-#
-#         mongo_changes_bulk = []
-#         for version in data["versions"]:
-#             orig_version_tags = version["data"].get("tags") or []
-#             version_tags = [tag for tag in orig_version_tags]
-#             if "deleted" not in version_tags:
-#                 version_tags.append("deleted")
-#
-#             if version_tags == orig_version_tags:
-#                 continue
-#
-#             update_query = {"id": version["id"]}
-#             update_data = {"$set": {"data.tags": version_tags}}
-#             mongo_changes_bulk.append(UpdateOne(update_query, update_data))
```

Comment:
Where is conversion of this logic?

Reply:
are we still using the mongo or we can use some postsql?